docs/mp/persistence.adoc

```diff
@@ -0,0 +1,1430 @@
+</dependency>
+----
+<1> The `scope` is `runtime`, indicating that the Oracle
+    Universal Connection Pool integration will be available on the
+    runtime classpath.
+
+==== Setting Up a Database Driver [[DS-Driver-Project-Setup]]
+
+===== Overview
+
+Regardless of which connection pool you use, at the lowest level JDBC
+database driver classes are responsible for making any connections to
+a relational database.  JDBC database driver classes are
+database-product-specific.
+
+Once you have decided upon a relational database product to use, and
+JDBC driver classes to use to connect to it,
+xref:../about/managing-dependencies.adoc[ensure your dependencies are
+managed], and then ensure that a `runtime`-scoped `<dependency>`
+element describing your JDBC driver classes is present as a child
+element of your project's `pom.xml` file's `<dependencies>` element.
+
+See the
+https://docs.oracle.com/en/java/javase/17/docs/api/java.sql/java/sql/package-summary.html[JDBC
+4.3 Specification] for more information about JDBC.
+
+Representative setups are described below.  The list of such setups is
+not exhaustive.
+
+===== Setting Up H2 [[DS-H2-Driver-Project-Setup]]
+
+====== Maven Coordinates (H2) [[DS-H2-Driver-Maven-Coordinates]]
+
+To include the http://www.h2database.com/html/main.html[H2 JDBC
+driver] classes in your Helidon MP application so your application can
+http://www.h2database.com/html/features.html#database_url[connect to
+an H2 database] (whether in-memory or persistent):
+
+* xref:../about/managing-dependencies.adoc[Ensure your dependencies are
+managed]
+
+* Ensure the following `<dependency>` element is present as a child
+  element of your project's `pom.xml` file's `<dependencies>` element:
++
+[source,xml]
+----
+<dependency>
+    <groupId>com.h2database</groupId>
+    <artifactId>h2</artifactId>
+    <scope>runtime</scope> <!--1-->
+</dependency>
+----
+<1> The `scope` is `runtime`, indicating that the H2
+    JDBC driver classes will be available on the runtime classpath.
+
+===== Setting Up Oracle JDBC [[DS-Oracle-Driver-Project-Setup]]
+
+====== Maven Coordinates (Oracle JDBC) [[DS-Oracle-Driver-Maven-Coordinates]]
+
+To include the
+https://docs.oracle.com/en/database/oracle/oracle-database/21/jajdb/index.html[Oracle
+JDBC driver classes] in your Helidon MP application so your
+application can
+https://docs.oracle.com/en/database/oracle/oracle-database/21/jjdbc/data-sources-and-URLs.html#GUID-EF07727C-50AB-4DCE-8EDC-57F0927FF61A[connect
+to an Oracle database]:
+
+* xref:../about/managing-dependencies.adoc[Ensure your dependencies are
+managed]
+
+* Read and understand
+  https://www.oracle.com/database/technologies/maven-central-guide.html[Developer's
+  Guide For Oracle JDBC 21c on Maven Central]
+
+* For a basic setup, ensure the following `<dependency>` element is
+  present as a child element of your project's `pom.xml` file's
+  `<dependencies>` element:
++
+[source,xml]
+----
+<dependency>
+    <!--
```

Comment:
This should go in the call-out below. Remember that the snippets can be copied to the clipboard, these comments will be included.

Reply:
That is exactly why I include them here. Would you prefer I not?

Reply:
Personally I don't think the snippet needs to carry additional information, though that's my opinion.

I don't recall any other dependency related snippets including comments on purpose.
If we do it for this dependency why aren't we doing for all dependencies ?

Reply:
I will remove all comments in dependency stanzas.